src/server/_query.py

```diff
@@ -185,7 +187,8 @@ def filter_pair(pair: TimePair, i) -> str:
         params[type_param] = pair.time_type
         if isinstance(pair.time_values, bool) and pair.time_values:
             return f"{type_field} = :{type_param}"
-        return f"({type_field} = :{type_param} AND {filter_integers(time_field, cast(Sequence[Union[int, Tuple[int,int]]], pair.time_values), type_param, params)})"
+        ranges = dates_to_ranges(pair.time_values)
```

Comment:
I tried this:
```
ranges = weeks_to_ranges(pair.time_values) if pair.is_week else days_to_ranges(pair.time_values)
```
but it fails on one of the integration tests, which has inputs that are neither valid weeks nor valid ranges (`0-999999`). `dates_to_ranges` has some extra validation to account for that

Reply:
`dates_to_ranges` works for that strange integration test case not because of extra validation, but because it passes back the input values through as-is when validation fails.  if you wrap the guts of `_to_ranges()` in a `try` (or at least wrap the part that populates the `intervals` list) and return the unmodified input on an exception, i think the test should pass.  you can also remove the try from `dates_to_ranges` after making that change.